src/software/ai/hl/stp/stp_test.cpp

```diff
@@ -67,42 +67,21 @@ TEST_F(STPTest, test_calculate_new_play_when_multiple_plays_valid)
     // deterministic because we have provided a seed for this test (in the setUp function)
 
     std::unique_ptr<Play> play;
-    for (int i = 0; i < 8; i++)
-    {
-        play = stp.calculateNewPlay(world);
-        EXPECT_TRUE(play);
-        EXPECT_EQ(play->getName(), MoveTestPlay::name);
-    }
-
-    play = stp.calculateNewPlay(world);
-    EXPECT_TRUE(play);
-    EXPECT_EQ(play->getName(), HaltTestPlay::name);
+    std::vector<std::string> actual_play_names;
```

Comment:
this doesn't seem within scope of this PR...

Reply:
So the order of the plays returned here is entirely random, but is compiler-specific (but still deterministic), so it changed when we went from gcc to clang. This felt like a much simpler way to write the test, but it's functionally the same.